src/Particle/ParticleSet.cpp

```diff
@@ -811,11 +811,18 @@ void ParticleSet::mw_donePbyP(const RefVectorWithLeader<ParticleSet>& p_list)
   for (ParticleSet& pset : p_list)
   {
     pset.coordinates_->donePbyP();
-    if (pset.SK && !pset.SK->DoUpdate)
-      pset.SK->UpdateAllPart(pset);
     pset.activePtcl = -1;
   }
 
+  if (p_leader.SK && !p_leader.SK->DoUpdate)
```

Comment:
Rename DoUpdate this reads as nonsense. I assume its a boolean.  SK would also really be better named structure_factor_, its very long lived and doesn't help the code be readable.

I do not like behavior control based on nullptr's but that's a bigger issue.

Reply:
I have a quick assess of DoUpdate. It is used to control SK update during p-by-p but the only wavefunction piece which need SK is k-space backflow which doesn't seem to have any p-by-p implementation. However, those functions may still needed for NLPP. I'd like to defer DoUpdate related changes as more reading will be needed.  Renaming SK also need to move SK out of public and add accessor. Needs to be done separately.